File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int size = s.size();
        int left = 0;
        int right = 0;
        int max_word = 0;
        string tmp;
        while (right < size)
        {
            if (tmp.find(s[right]) != string::npos)
            {
                tmp.erase(0, 1);
                left++;
            }
            else
            {
                tmp += s[right];
                max_word = max(max_word, right - left + 1);
                right++;
            }
        }
        return max_word;
    }
};

File: Data Structures & Algorithms/binary-tree-diameter/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def diameterOfBinaryTree(self, root: Optional[TreeNode]) -> int:
        self.delm = 0
        def dfs(head: Optional[TreeNode]) -> int:
            if not head:
                return 0
            left = dfs(head.left)
            right = dfs(head.right)
            self.delm = max(self.delm, left + right)
            return 1 + max(left, right)
        dfs(root)
        return self.delm